src/mamba2_torch/modeling/modeling_mamba2.py

```diff
@@ -783,48 +805,89 @@ def set_input_embeddings(self, new_embeddings):
         return self.backbone.set_input_embeddings(new_embeddings)
 
     def _update_model_kwargs_for_generation(
-            self, outputs: ModelOutput, model_kwargs: Dict[str, Any], **kwargs
+            self, outputs: ModelOutput, model_kwargs: Dict[str, Any], num_new_tokens: int = 1, **kwargs
     ) -> Dict[str, Any]:
         model_kwargs["cache_params"] = outputs.get("cache_params", None)
+
+        if (
+                model_kwargs.get("use_cache", True)
+                and "cache_position" in model_kwargs
+                and model_kwargs["cache_position"] is not None
+        ):
+            model_kwargs["cache_position"] = model_kwargs["cache_position"][-1:] + num_new_tokens
```

Comment:
We could likely ignore `cache_position` as it's used for the cache to be torch compile compatible but since the mamba2 kernel can't really compile either way...